packages/components/src/custom-select-control/test/index.js

```diff
@@ -87,4 +88,127 @@ describe( 'CustomSelectControl', () => {
 			screen.getByRole( 'button', { name: 'Custom select' } )
 		).toHaveTextContent( 'Hint' );
 	} );
+
+	it( 'Initial value should be replaced when a new item is selected', async () => {
+		const user = userEvent.setup();
+		const mock = jest.fn();
+
+		render(
+			<CustomSelectControl
+				onChange={ ( { selectedItem } ) => mock( selectedItem ) }
+				options={ options }
+				__nextUnconstrainedWidth
+			/>
+		);
+
+		const currentValue = screen.getByRole( 'button' );
+
+		expect( currentValue ).toHaveTextContent( /violets/i );
+
+		await user.click( currentValue );
+
+		const newValue = screen.getByRole( 'option', {
+			name: /poppy/i,
+		} );
+
+		await user.click( newValue );
+
+		expect( currentValue ).toHaveTextContent( /poppy/i );
+		expect( mock ).toHaveBeenCalled();
+	} );
+
+	it( 'Should be able to change selection using keyboard', async () => {
+		const user = userEvent.setup();
+
+		render(
+			<CustomSelectControl options={ options } __nextUnconstrainedWidth />
+		);
+
+		const currentValue = screen.getByRole( 'button' );
+		expect( currentValue ).toHaveTextContent( /violets/i );
+
+		await user.tab();
+		expect( currentValue ).toHaveFocus();
+
+		await user.keyboard( '{enter}' );
+		await user.keyboard( '{arrowdown}' );
+		await user.keyboard( '{enter}' );
+
+		expect( currentValue ).toHaveTextContent( /crimson clover/i );
+	} );
+
+	it( 'Should keep current selection if dropdown is closed without changing selection', async () => {
+		const user = userEvent.setup();
+
+		render(
+			<CustomSelectControl options={ options } __nextUnconstrainedWidth />
+		);
+
+		const currentValue = screen.getByRole( 'button' );
+		expect( currentValue ).toHaveTextContent( /violets/i );
+
+		await user.tab();
+		await user.keyboard( '{enter}' );
+		expect( screen.getByRole( 'listbox' ) ).toBeVisible();
+		expect( screen.getByRole( 'listbox' ) ).toHaveAttribute(
+			'aria-hidden',
+			'false'
+		);
+		await user.keyboard( '{escape}' );
+		expect( screen.queryByRole( 'listbox' ) ).not.toBeInTheDocument();
```

Comment:
Should we await for this to happen? Theoretically, it could get flaky, because there's some ms required between handling the `escape` key press and the disappearance of the element. 

We could use `waitForElementToBeRemoved()`: https://testing-library.com/docs/guide-disappearance/#waiting-for-disappearance

Reply:
If we wait for it to be removed via the method you shared, this error occurs: 

```
The element(s) given to waitForElementToBeRemoved are already removed. waitForElementToBeRemoved requires that the element(s) exist(s) before waiting for removal.
```